File: models/Gpt-3.5-turbo/py/code/additional_problems/Programmers_and_Artists.py

# Read input
a, b, n = map(int, input().split())
applicants = []
for _ in range(n):
    x, y = map(int, input().split())
    applicants.append((x+y, x, y))

# Sort applicants in descending order of total skills
applicants.sort(reverse=True)

# Counters for the number of hired programmers and artists
p_count = 0
a_count = 0

# Sum of skills
max_skill_sum = 0

# Iterate through applicants
for _, x, y in applicants:
    # Check if the applicant has higher programming skills and we need more programmers
    if p_count < a and x >= y:
        max_skill_sum += x
        p_count += 1
    # Check if the applicant has higher artistic skills and we need more artists
    elif a_count < b and y > x:
        max_skill_sum += y
        a_count += 1

# Print the maximum sum of skills
print(max_skill_sum)

File: models/Gpt-3.5-turbo/cpp/code/problems/additional_problems/Programmers_and_Artists.cpp

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

struct Applicant {
    int programming_skill;
    int artistic_skill;
};

bool compare(Applicant a, Applicant b) {
    // Sort in descending order based on sum of skills
    return (a.programming_skill + a.artistic_skill) > (b.programming_skill + b.artistic_skill);
}

int main() {
    int a, b, n;
    cin >> a >> b >> n;

    vector<Applicant> applicants(n);
    for (int i = 0; i < n; ++i) {
        cin >> applicants[i].programming_skill >> applicants[i].artistic_skill;
    }

    sort(applicants.begin(), applicants.end(), compare);

    long long max_skill_sum = 0;
    for (int i = 0; i < a; ++i) {
        max_skill_sum += applicants[i].programming_skill;
    }
    for (int i = a; i < a + b; ++i) {
        max_skill_sum += applicants[i].artistic_skill;
    }

    cout << max_skill_sum << endl;

    return 0;
}
